visionmetrics/key_value_pair/key_value_pair_eval.py

```diff
@@ -0,0 +1,262 @@
+        # Reserve one class to catch cases where predictions are not in the set of expected classes
+        class_map[OUT_OF_DISTRIBUTION_ENUM_KEY] = len(class_map)
+        return class_map
+
+    def _assign_key_metric_map_values(self, key: str, metric_name: str, metric_args: dict, class_map: dict = None):
+        """
+        Assigns values for the metric_name, metric_args, and class_map for a given key in the object-level key_metric_map dictionary.
+        Args:
+            key: string of the key name, which should be identical to the key that will be evaluated in the prediction and target dictionaries.
+            metric_name: string of the metric name as defined in visionmetrics (e.g., 'classification.MulticlassAccuracy'), which
+                         should be among those specified in key_value_pair_eval_base.SupportedKeyWiseMetric.
+            metric_args: dictionary of args to pass in as keyword arguments to the initialization function of the metric.
+            class_map (optional): dictionary mapping string class names to integer class indices for this key.
+        """
+        if key not in self.key_metric_map:
+            self.key_metric_map[key] = {}
+        self.key_metric_map[key]["metric_name"] = metric_name
+        self.key_metric_map[key]["metric_args"] = metric_args
+        if class_map:
+            self.key_metric_map[key]["class_map"] = class_map
+
+    def _populate_key_metric_map(self, key: str, key_schema: dict, key_trace: list):
+        """
+        Recursive function that populates the object-level key_metric_map dictionary. For a given key and key_schema in JSON Schema format,
+        the function populates the metric_name, metric_args, and preprocessor values in the dictionary.
+        Note: The recursive behavior currently only occurs for JSONSchemaKeyType.Object key types, not arrays.
+        Args:
+            key: string of the key name, which should be identical to the key that will be evaluated in the prediction and target dictionaries.
+            key_schema: dictionary in JSON schema format specifying the expected schema for the prediction and target dictionaries to be used in evaluation.
+            key_trace: list of strings of key names that traces the path to the current key in the key-value pair prediction/target object (not in the schema).
+        """
+        # Use text as the default metric for all keys
+        self._assign_key_metric_map_values(key=key,
+                                           metric_name=SupportedKeyWiseMetric.Caption_AzureOpenAITextModelCategoricalScore,
+                                           metric_args={"endpoint": self.endpoint, "deployment_name": self.deployment_name})
+        if key_schema["type"] == JSONSchemaKeyType.String:
+            if "enum" in key_schema:
+                class_map = self._get_enum_class_map(key_schema["enum"])
+                self._assign_key_metric_map_values(key=key,
```

Comment:
nit: instead of repeated calls to `self._assign_key_metric_map_values`, may initiate empty mapping dict at the top and fill it as per every IF/ELSE condition and call `self._assign_key_metric_map_values(**mapping)` at the end. 
If mapping is empty after all IF/ELSE, then initialize it with the default metric